plugins/org.jkiss.dbeaver.data.transfer/src/org/jkiss/dbeaver/tools/transfer/stream/StreamTransferConsumer.java

```diff
@@ -1065,14 +1066,22 @@ public void writeBinaryData(@NotNull DBDContentStorage cs) throws IOException {
                                 writer.write(binaryString);
                                 break;
                             }
+                            break;
                         }
+                        case BINARY:
                         default: {
-                            // Binary stream
-                            try (Reader reader = new InputStreamReader(stream, cs.getCharset())) {
-                                IOUtils.copyText(reader, writer);
+                            byte[] readBuffer = new byte[1000];
+                            for (; ; ) {
+                                int count = stream.read(readBuffer);
+                                if (count <= 0) {
+                                    break;
+                                }
+                                String content = new String(readBuffer, 0, count, settings.getOutputEncoding());
```

Comment:
You should be using `cs.getCharset()` instead.

Reply:
I guess it was a mistake to use `cs.getCharset()` before. As I see there is need to take charset from params, cause we have such properties in configuration page

Reply:
But this is not correct. You are reading data from `cs` and should use its charset, not the output charset. The output charset is already used by the underlying stream.